introspection-engine/connectors/sql-introspection-connector/src/introspection_map.rs

```diff
@@ -40,11 +41,45 @@ impl<'a> IntrospectionMap<'a> {
         relation_names::introspect(input, &mut map);
         position_inline_relation_fields(sql_schema, &mut map);
         position_m2m_relation_fields(sql_schema, &mut map);
+        populate_top_level_names(sql_schema, prisma_schema, &mut map);
 
         map
     }
 }
 
+fn populate_top_level_names<'a>(
+    sql_schema: &'a sql::SqlSchema,
+    prisma_schema: &'a psl::ValidatedSchema,
+    map: &mut IntrospectionMap<'a>,
+) {
+    for table in sql_schema
+        .table_walkers()
+        .filter(|t| !helpers::is_prisma_join_table(*t))
+    {
+        let name = map
+            .existing_models
+            .get(&table.id)
+            .map(|id| prisma_schema.db.walk(*id))
+            .map(|m| m.name())
+            .unwrap_or_else(|| table.name());
```

Comment:
I am wondering if we should take the names after sanitation into account. For example `ab&c` and `ab_c` would clash only after sanitation. That's really an edge case though, I'm completely OK with waiting until it actually happens.

Reply:
We look into duplicates in the PSL namespace. You can definitely rename your model to have the same name as an incoming table in the introspection. It is rare, but annoying if it happens.

Reply:
Um, the point about sanitation is valid though... Then we'd need to store cows here... I'll think about doing that during the commute.